tools/ddrgen/src/scanners/dwarf/DwarfScanner.cpp

```diff
@@ -191,7 +193,10 @@ DwarfScanner::blackListedDie(Dwarf_Die die, bool *dieBlackListed)
 		/* If the decl_file matches an entry in the file table not beginning with "/usr/",
 		 * then we are interested in it. Also filter out decl file "<built-in>".
 		 */
-		if ((NULL == strstr(_fileNamesTable[declFile - 1], "<built-in>"))
+		if (0 == declFile) {
+			/* What happens when there is no file? */
```

Comment:
I took a look at the DWARF documentation, and it says "The value 0 indicates that no source file has been specified."

So the decl_file attribute being 0 is supported, so I think we need this check for that case.